test/langtools/tools/javac/reflect/SwitchExpressionTest2.java

```diff
@@ -0,0 +1,633 @@
+import java.lang.runtime.CodeReflection;
+
+/*
+ * @test
+ * @enablePreview
+ * @build SwitchExpressionTest2
+ * @build CodeReflectionTester
+ * @run main CodeReflectionTester SwitchExpressionTest2
+ */
+public class SwitchExpressionTest2 {
```

Comment:
I'm a bit confused: the core of the fix is to add a conversion to the selector type is the case label is constant, right? But I don't see any `conv` node anywhere in the test (except for a single one where an explicit cast is used). Is the new code exercised in this test?

Reply:
The core of the changes is to test switch expression when case is constant, in the process we found that we miss conversion of the constant to the the type of the selector expression, we added that. I agree that we should add tests that focus more on conversion.

Reply:
`caseConstantThrow` shows how we are converting constant label to the selector expression's type.

Reply:
I see - I was looking for `convert` but it's true that there are some unboxing conversions in there